fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -338,6 +338,11 @@ class FullyShardedDataParallel(nn.Module):
             rank 0 and return empty dict non-rank 0, which allow FullyShardedDataParallel to
             skip the GPU -> CPU copy on non-rank 0 altogether and prevent OOM.
             Default: False
+        optimize_backward_concat (bool):
+            If True, only let backward pass propagate to self.params, which will
+            invoke the _post_backward_hook() and concat() op, when self._require_backward_grad_sync
+            is True (e.g. last microbatch)
+            NOTE: this likely will incur more GPU memory usage
```

Comment:
Could you explain why there will be more GPU memory usage?

Reply:
hi @awgu, currently by testing results it shows the GPU memory overhead could be non-trivial (20% of 80G), we will follow up on reducing the memory usage
<img width="519" alt="Screenshot 2024-05-15 at 10 39 19 AM" src="https://github.com/facebookresearch/fairscale/assets/6046382/d27c7c99-e28d-43cd-9b6b-cee3ab467bb9">
<img width="529" alt="Screenshot 2024-05-15 at 10 40 18 AM" src="https://github.com/facebookresearch/fairscale/assets/6046382/197f0f93-42f5-4fb3-a116-782ebd0175c6">